### PR TITLE
Feature/general improvements

### PR DESCRIPTION
**Pull request by lrascao**
This set of commits are general quality of life minor improvements, tests were failing on MacOS and should be passing now. 
This is a WIP as it's waiting on https://github.com/rusterlium/erlang-cargo/pull/6, a second PR after that one and a Hex version cut.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,4 +1,8 @@
-{deps, [cargo]}.
+{deps, [
+    {cargo, ".*",
+        {git, "https://github.com/lrascao/erlang-cargo.git",
+            {branch, "feature/target_dir"}}}
```
Comment (lrascao) [inline rebar.config]: this branch is perched on top of https://github.com/rusterlium/erlang-cargo/pull/6 and will be PR'd after it

**Comment by filmor** [comment]
Ah, sorry, I didn't see the part about you waiting for a Hex release, will do that tomorrow.

**Comment by lrascao** [comment]
I guess we can wait until this one passes review

**Comment by lrascao** [comment]
i'll try and squeeze in some time this week so it can promoted out of WIP and start the review

**Comment by lrascao** [comment]
@filmor this is now ready for review, if approved, before merging we can cut a new hex version of `erlang-cargo` and fixup 95aadb3

File: src/rebar3_cargo_compile_prv.erl [hunk 2]
```diff
@@ -154,8 +149,17 @@ write_header(App, NifLoadPaths) ->
         "-endif.\n"
     ],
 
-    OutDir = rebar_app_info:dir(App),
-    OutPath = filename:join([OutDir, "src", "crates.hrl"]),
+    OutDir0 = rebar_app_info:dir(App),
+    IncludeDir = filename:join(OutDir0, "include"),
+    SrcDir = filename:join(OutDir0, "src"),
+    % prefer the include dir to generate crates.hrl to (if it exists)
+    OutDir = case filelib:is_dir(IncludeDir) of
```
Comment (filmor) [inline src/rebar3_cargo_compile_prv.erl]: No, the `crates.hrl` is not a "public" header file.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: agree, and it should be git ignored by any project using it. But if it's getting generated somewhere might as well be `include` (if possible), tipically only `.erl` files live in `src`

File: ct/rebar3_cargo_SUITE_data/test_port_app/src/complex1.erl [hunk 3]
```diff
@@ -28,6 +28,8 @@ call_port(Msg) ->
 init(ExtPrg) ->
     register(complex, self()),
     process_flag(trap_exit, true),
+    io:format("ExtPrg: ~p\n",
+              [ExtPrg]),
```
Comment (lrascao) [inline ct/rebar3_cargo_SUITE_data/test_port_app/src/complex1.erl]: missed this, will delete it

File: src/rebar3_cargo_compile_prv.erl (hunk 2 above)
Comment (filmor) [reply src/rebar3_cargo_compile_prv.erl]: The convention seems to be to have internal headers in `src`.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: ok, i can move this commit out of this PR context, another way would be to introduce an option for the target dir of `crates.hrl` that defaults to `src`, thoughts?

**Comment by lrascao** [comment]
@filmor just checking, is something blocking this or just not enough bandwidth atm?

File: src/rebar3_cargo_compile_prv.erl (hunk 2 above)
Comment (filmor) [reply src/rebar3_cargo_compile_prv.erl]: I still stand by this, please move this change out.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: 7dad8595 has been removed

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: 1e489afe that contained this change has been dropped

**Comment by filmor** [comment]
Sorry for the lag, looks good, I'll go over it once more on the weekend and merge.

**Review by filmor — APPROVED** [review]

**Pull request merged**